### PR TITLE
docs: correct some details in high level design doc

### PR DESCRIPTION
**Pull request by amyangfei**
## Description

Correct some details in high level design doc, including using relative link path, using correct variable name.

## Related Issue(s)

https://github.com/vitessio/vitess/pull/12071

## Checklist

-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by deepthi — APPROVED** [review]

**Review by GuptaManan100 — APPROVED** [review]
changes look good to me! Thank-you for catching and fixing them!

**Review by frouioui — APPROVED** [review]
Looks good to me, thank you for contributing this! 🙏🏻

**Comment by amyangfei** [comment]
The CI workflow requires approval from a maintainer, who will help, many thanks 😄

**Pull request closed without merging**
